### PR TITLE
[Fleet][dev] 06-15-preview

### PR DESCRIPTION
**Pull request by zman-ms**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [x] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Comment by serbrech** [comment]
<html>
<body>


Rule | Message
-- | --
❌ 1011 - AddingResponseCode | The new version adds a response code '202'.New: fleet/preview/2022-09-02-preview/fleets.json#L358:11
❌ 1011 - AddingResponseCode | The new version adds a response code '202'.New: fleet/preview/2023-03-15-preview/fleets.json#L361:11
❌ 1011 - AddingResponseCode | The new version adds a response code '202'.New: fleet/preview/2023-03-15-preview/fleets.json#L785:11
❌ 1044 - XmsLongRunningOperationChanged | The new version has a different 'x-ms-longrunning-operation' value than the previous one.New: fleet/preview/2022-09-02-preview/fleets.json#L307:7Old: fleet/preview/2022-09-02-preview/fleets.json#L307:7
❌ 1044 - XmsLongRunningOperationChanged | The new version has a different 'x-ms-longrunning-operation' value than the previous one.New: fleet/preview/2023-03-15-preview/fleets.json#L310:7Old: fleet/preview/2023-03-15-preview/fleets.json#L310:7
❌ 1044 - XmsLongRunningOperationChanged | The new version has a different 'x-ms-longrunning-operation' value than the previous one.New: fleet/preview/2023-03-15-preview/fleets.json#L724:7Old: fleet/preview/2023-03-15-preview/fleets.json#L706:7

These breaking changes are **known**. We are adding an async patch implementation to the api. ManagedIdentity portal integration is built for patch async. we want to take this breaking change now in preview.
this will cause an SDK breaking change for the 06-15-preview sdks, but does not break at the API level for all existing versions.
the SDKs will not be re-generated.

Typespec does not provide a good way to version this change, so we'd like to apply it to all for the sake of future maintenance of the service.

File: specification/containerservice/Fleet.Management/helpers.tsp [hunk 1]
```diff
@@ -24,96 +24,52 @@ alias IfNoneMatchParameters<T extends Azure.ResourceManager.Foundations.ArmResou
...
   "x-ms-long-running-operation-options",
   {
-    `final-state-via`: "location",
+    "final-state-via": "location",
   }
 )
 @post
 op ArmResourceNoRequestBodyActionAsync<
   TResource extends Azure.ResourceManager.Foundations.ArmResource,
-  TResponse extends TypeSpec.Reflection.Model,
```
Comment (serbrech) [inline specification/containerservice/Fleet.Management/helpers.tsp:61]: revert this line change

File: specification/containerservice/Fleet.Management/package.json
Comment (serbrech) [inline specification/containerservice/Fleet.Management/package.json:1]: remove file from PR

File: specification/containerservice/Fleet.Management/package-lock.json
Comment (serbrech) [inline specification/containerservice/Fleet.Management/package-lock.json:1]: remove file from PR

File: specification/containerservice/Fleet.Management/tspconfig.yaml
Comment (serbrech) [inline specification/containerservice/Fleet.Management/tspconfig.yaml:1]: revert this `git checkout specification/containerservice/Fleet.Management/tspconfig.yaml`

File: specification/containerservice/Fleet.Management/.npmrc
Comment (serbrech) [inline specification/containerservice/Fleet.Management/.npmrc:1]: remove this file

File: package-lock.json
Comment (serbrech) [inline package-lock.json:1]: revert this

File: specification/containerservice/Fleet.Management/main.tsp [hunk 2]
```diff
@@ -19,12 +19,15 @@ namespace Microsoft.ContainerService;
...
-#suppress "@azure-tools/typespec-azure-core/documentation-required" "https://github.com/Azure/typespec-azure/issues/3107"
 enum Versions {
   @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
   v2022_09_02_preview: "2022-09-02-preview",
 
   @useDependency(Azure.Core.Versions.v1_0_Preview_2)
   @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
   v2023_03_15_preview: "2023-03-15-preview",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
```
Comment (serbrech) [inline specification/containerservice/Fleet.Management/main.tsp:30]: are there newer versions of these dependencies?

File: specification/containerservice/Fleet.Management/updaterun.tsp [hunk 3]
```diff
@@ -21,50 +21,49 @@ namespace Microsoft.ContainerService;
...
+    @key("updateRunName")
+    @segment("updateRuns")
+    @pattern("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+    @minLength(1)
+    @maxLength(50)
+    @path
+    @visibility("create", "read")
+    name: string;
+    ...EntityTag;
 }
```
Comment (serbrech) [inline specification/containerservice/Fleet.Management/updaterun.tsp:34]: why is the indentation changing?

Comment (zman-ms) [reply specification/containerservice/Fleet.Management/updaterun.tsp:34]: Ran tsp format and fixed.

File: specification/containerservice/Fleet.Management/.gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+node_modules
+output
+tspconfig.yaml
```
Comment (serbrech) [inline specification/containerservice/Fleet.Management/.gitignore:3]: Shouldn't ignore tspconfig.yaml, but should have the same as what's currently in the main branch.

File: specification/containerservice/Fleet.Management/examples/2022-09-02-preview/Fleets_PatchTags.json [hunk 5]
```diff
@@ -46,6 +46,12 @@
           "provisioningState": "Succeeded"
         }
       }
+    },
+    "202": {
+      "headers": {
+        "Retry-After": "20",
+        "Location": "East US"
```
Comment (serbrech) [inline specification/containerservice/Fleet.Management/examples/2022-09-02-preview/Fleets_PatchTags.json:53]: location is not East US, it's the operationStatus url. 
See Delete example :

```
  "Location": "https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ContainerService/locations/eastus/operations/00000000-0000-0000-0000-000000000000?api-version=2022-02-01"
  ```

**Review by serbrech — APPROVED** [review]

**Pull request closed without merging**